doxygen2man/doxygen2man.c

```diff
@@ -498,10 +500,13 @@ static char *allcaps(const char *name)
 	static char buffer[4096] = {'\0'};
 	size_t i;
 
-	for (i=0; i< strlen(name); i++) {
-		buffer[i] = toupper(name[i]);
+	if (name) {
+		size_t len = strnlen(name, 4096);
+		for (i=0; i< len; i++) {
+			buffer[i] = toupper(name[i]);
+		}
+		buffer[len] = '\0';
```

Comment:
BTW you can also shave off the `len` declaration by using `buffer[i] = 0` here (if this is your thing). :)